internal/controller/pkg/revision/dependency.go

```diff
@@ -205,7 +202,7 @@ func (m *PackageDependencyManager) Resolve(ctx context.Context, pkg runtime.Obje
 
 // RemoveSelf removes a package from the lock.
 func (m *PackageDependencyManager) RemoveSelf(ctx context.Context, pr v1beta1.PackageRevision) error {
-	prRef, err := name.ParseReference(pr.GetSource())
+	prRef, err := name.ParseReference(pr.GetSource(), name.WithDefaultRegistry(""))
```

Comment:
My understanding of this change is that, were this option not specified, `go-containerregistry` will default to `index.docker.io` (src: https://github.com/google/go-containerregistry/blob/master/pkg/name/options.go#L37)

So we are effectively deferring the job of picking a default domain for an unqualified image tag to the code that ultimately does the pull, which I think in this case is going to be the package manager. What I don't know without looking deeper is what part of the package management system or libraries will pick the domain to use in the request?

Reply:
There are three places where we do this:
1. Getting the digest of the current package image: https://github.com/crossplane/crossplane/blob/dce235c83c2990bb8a521c86223efd07c63001a2/internal/controller/pkg/manager/revisioner.go#L62
2. Actually downloading the package into the cache: https://github.com/crossplane/crossplane/blob/dce235c83c2990bb8a521c86223efd07c63001a2/internal/controller/pkg/revision/imageback.go#L78
3. Getting all tags for a dependency: https://github.com/crossplane/crossplane/blob/dce235c83c2990bb8a521c86223efd07c63001a2/internal/controller/pkg/resolver/reconciler.go#L220

Reply:
SGTM